### PR TITLE
feat/LIVE-3138 Introduce hasDevFirmware flag into DeviceInfo

### PR DESCRIPTION
**Pull request by juan-cortes**
### 📝 Description

In order to provide a better UX and prevent wasting time debugging issues that are not really there (such as connectivity issues that arise from using untested firmware or unsupported firmware for the current provider). This PR introduces a best-effort development firmware version detection flag into the `DeviceAction` response.

We need the hardcoded list of suffixes because it is not clear that firmware will never user letters in the versioning, that's what I mean by best-effort.

It could then be used in the UI to provide feedback to the user or customize the error handling/tracking.

### ❓ Context

- **Impacted projects**: `ledger-live-common` 
- **Linked resource(s)**: `` 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo
No visuals here, covered by tests.

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/src/hw/isDevFirmware.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * In order to mitigate users on odd firmware versions reporting
+ * connectivity errors when interacting with the app. Attempt to detect
+ * those versions and set a flag for UI.
+ */
+const isDevFirmware = (seVersion: string | undefined): boolean => {
+  if (!seVersion) return false;
+
+  const knownDevSuffixes = ["lo", "rc", "il", "tr"]; // FW can't guarantee non digits in versions
+  const pattern = new RegExp(`.+-(${knownDevSuffixes.join("|")})\\d{0,}`);
```
Comment (gre) [inline libs/ledger-live-common/src/hw/isDevFirmware.ts:10]: that part is overkill i think.
is there a way NOT to use a regexp? RegExp construction may be slow in some cases.

would it work to do something like `seVersion.includes("-rc")` 🤔  ultimately, isn't the presence of a `-` a dev firmware? (maybe i'm wrong)

Comment (juan-cortes) [reply libs/ledger-live-common/src/hw/isDevFirmware.ts:10]: No, @rgeslain-ledger said that's the only route. I don't think it's overkill though, we only run this when we run the `getDeviceInfo`, not on every frame or render.

Comment (gre) [reply libs/ledger-live-common/src/hw/isDevFirmware.ts:10]: anyway, this should work

```js
const knownDevSuffixes = ["lo", "rc", "il", "tr"];
knownDevSuffixes.some(s => seVersion.includes("-"+s))
```

Comment (juan-cortes) [reply libs/ledger-live-common/src/hw/isDevFirmware.ts:10]: Sure ... 🤷🏼

**Review by Justkant — APPROVED** [review]

**Comment by gre** [comment]
need to regen documentation apparently

**Review by hedi-edelbloute — APPROVED** [review]

**Comment by Justkant** [comment]
We also probably need a changeset

**Comment by juan-cortes** [comment]
Changeset and docs ready

**Pull request merged**